sourcegraph/sourcegraph
# storm: migrate feature flags to Apollo Client

## Context

Based on [the experimental branch](https://github.com/sourcegraph/sourcegraph/pull/47888). This PR goes further and removes the `FeatureFlagClient` altogether and replaces its functionality with the logic in React hook. It's possible because now the cache is managed by Apollo, and we only add a layer on top of it that ensures that we refetch values once `cacheTTL` elapses.

This change allows us to skip evaluate feature flag API requests in Storm routes if feature flag values are preloaded!

## Test plan

1. Enable Storm in experimental features.
2. Load the home page.
3. Notice that the `search-ownership` and `plg-enable-add-codehost-widget` feature flags are not fetched separately. They are loaded as a part of the `SearchPageQuery`, and the `useFeatureFlag` hook reuses cached values.

## App preview:

- [Web](https://sg-web-vb-storm-4.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
